### ISSUE TITLE
Upgrading db image to MySQL 8 causes plugin load issue

### ISSUE THREAD

**Issue by markshust** [markshust/docker-magento#393]
**Steps To Reproduce**

1. Do manual setup of new project
2. Update `docker-compose.yml` file's `db` service to use `image: mysql:8.0.23` before starting install
3. Install the project
4. Run `bin/mysqldump > magento.sql` to create backup of database

**Expected Result**

Running script creates backup of MySQL database.

**Actual Result**

Running script outputs error:

```
mysqldump: Got error: 1045: "Plugin caching_sha2_password could not be loaded: /usr/lib/x86_64-linux-gnu/mariadb19/plugin/caching_sha2_password.so: cannot open shared object file: No such file or directory" when trying to connect
```

**Comment by rangerz** [comment]
MySQL 8 changed their password authentication method.
Using mysql_native_password plugin may solve this issue.
https://dev.mysql.com/doc/refman/8.0/en/native-pluggable-authentication.html

`bin/mysql` also not work for MySQL 8, but I found a way to use it.
`docker-compose exec db mysql`

Follow the step to setup mysql_native_password for `magento` user
```
docker-compose exec db mysql -u root -pmagento
mysql> ALTER USER 'magento'@'%' IDENTIFIED WITH mysql_native_password BY 'magento';
```
And then `bin/mysql` and `bin/mysqldump` can work fine for MySQL 8

**Comment by markshust** [comment]
Thanks, we did have this ticket just come in as well https://github.com/markshust/docker-magento/issues/396 -- I'm not sure if we'll face the same issue with MariaDB or not.

**Comment by EmperorKonstantin** [comment]
Seems like there's a fix for this here:

https://github.com/passbolt/passbolt_docker/issues/103#issuecomment-518228846

**Comment by markshust** [comment]
Thank you, I did try that before, I swear... but it didn't work. I'll attempt again 😀

**Comment by markshust** [comment]
Closing this out as an old ticket, as we just upgraded to MariaDB 10.4. If there is an issue with that image, please open a new ticket with the details, and we can take it from there.
